### DISCUSSION TITLE
How should I go about adding "favorite films" to a user account? (and also comments)

### DISCUSSION THREAD

**Discussion by velllu** [pocketbase/pocketbase#2968] [Q&A] [answered]
I want each user to have a list of favorite films, how can this be achieved?
I can't really use a relation, because the films are served from another source, so I don't have a collection with all the films.

Also, I would like to also have the ability of making users comment the films

Many thanks to anyone that replies, I am new with relational databases

**Comment by ganigeorgiev** [comment, marked as answer]
I don't think there is a universal answer to your question as it will depend on a lot of things, including the API of the external service you are planning to use.

In general, if the films are not imported in PocketBase, then you can't use a `relation` field and if you want to store some references (eg. film id) you'll have to store it as regular `text` field.

For example, you can create a base collection `favoriteFilms` with 2 fields:
- **user** - nonempty single `relation` field to the `users` collection
- **film** - nonempty regular `text` field, containing the id (or other unique string identifier) for the film

Then to fetch all favorite film relations for the current authenticated user, you can use [indirect expand](https://pocketbase.io/docs/expanding-relations/#indirect-expand) or send separate request:

  ```js
  // in the Admin UI you can use a List/Search API rule like "user = @request.auth.id"
  // ---

  // authenticate
  await pb.collection("users").authWithPassword("test@example.com", "1234567890")

  // fetch all favoriteFilms relations for the current authenticated user
  const favorites = await pb.collection("favoriteFilms").getFullList()
  ```

But keep in mind that the `film` field depends on external source and you'll have to imlement some sort of validation on your own if needed (eg. with a [`OnRecordBefore*Request` hook](https://pocketbase.io/docs/event-hooks/#record-crud-api-hooks); _with the upcoming v0.17.0 there will be option to bind to the hooks with JS too_).
